python/lsst/daf/butler/json.py

```diff
@@ -42,7 +44,7 @@ class SupportsSimple(Protocol):
     serialization using "simple" methods names.
     """
 
-    _serializedType: Type
+    _serializedType: type[BaseModel]
```

Comment:
Should actually by `ClassVar[type[BaseModel]]`, I suspect.

Aside: this is correct for the code as it is now, but I think we really want to support things that are recognized by Pydantic other than `BaseModel` (like `typing.TypedDict`) as "simple" objects in the future.  I think that would involve making this into:

```
class SupportsSimple(Protocol, Generic[_T]):

    _adapter: ClassVar[pydantic.TypeAdapter[_T]]
```

Reply:
Maybe, but `TypeAdapter` doesn't support `model_dump_json` which is what the code relies on (it's called `dump_json` with `TypeAdapter`).

I'll do your `ClassVar` fix.